strawberry/extensions/runner.py

```diff
@@ -25,65 +23,45 @@ class SchemaExtensionsRunner:
     def __init__(
         self,
         execution_context: ExecutionContext,
-        extensions: Optional[
-            List[Union[Type[SchemaExtension], SchemaExtension]]
-        ] = None,
+        extensions: Optional[List[SchemaExtension]] = None,
     ) -> None:
         self.execution_context = execution_context
-
-        if not extensions:
-            extensions = []
-
-        init_extensions: List[SchemaExtension] = []
-
-        for extension in extensions:
-            # If the extension has already been instantiated then set the
-            # `execution_context` attribute
-            if isinstance(extension, SchemaExtension):
-                extension.execution_context = execution_context
-                init_extensions.append(extension)
-            else:
-                init_extensions.append(extension(execution_context=execution_context))
-
-        self.extensions = init_extensions
+        self.extensions = extensions if extensions is not None else []
 
     def operation(self) -> OperationContextManager:
-        return OperationContextManager(self.extensions)
+        return OperationContextManager(self.extensions, self.execution_context)
 
     def validation(self) -> ValidationContextManager:
-        return ValidationContextManager(self.extensions)
+        return ValidationContextManager(self.extensions, self.execution_context)
 
     def parsing(self) -> ParsingContextManager:
-        return ParsingContextManager(self.extensions)
+        return ParsingContextManager(self.extensions, self.execution_context)
 
     def executing(self) -> ExecutingContextManager:
-        return ExecutingContextManager(self.extensions)
+        return ExecutingContextManager(self.extensions, self.execution_context)
+
+    @classmethod
+    def _implments_get_rseults(cls, extension: SchemaExtension) -> bool:
+        """Whether the extension implements get_results."""
+        return type(extension).get_results is not SchemaExtension.get_results
```

Comment:
Does anyone know whether these explicit checks might be slower than just calling the method and catching possible `NotImplementedError` exceptions (which we could let the unimplemented method throw by default).

Reply:
Actually seems thats a downgrade.
https://perfpy.com/743

Maybe we can cache them though.